source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -2196,6 +2196,211 @@ def test_add_new_trip_present_in_navitia_with_a_shift(self):
+        assert response['journeys'][0]['sections'][0]['departure_date_time'] == '20120615T180100'
+        assert response['journeys'][0]['sections'][0]['base_arrival_date_time'] == '20120615T180102'
+        assert response['journeys'][0]['sections'][0]['arrival_date_time'] == '20120616T010102'
+
+        # vjB circulates the day before at 18:01:00 and arrival at 18:01:02
+        response = self.query_region(ba_14T18_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120614T180100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120614T180102'
+        assert response['journeys'][0]['sections'][0]['display_informations']['name'] == 'B'
+
+        # vjB circulates the day after at 18:01:00 and arrival at 18:01:02
+        response = self.query_region(ba_16T18_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120616T180100'
+        assert response['journeys'][0]['arrival_date_time'] == '20120616T180102'
+        assert response['journeys'][0]['sections'][0]['display_informations']['name'] == 'B'
+
+        # Disruption is modified with first station on detour and delay so that there is no more pass midnight
+        self.send_mock(
+            "vjB",
+            "20120615",
+            'modified',
+            [
+                UpdatedStopTime(
+                    "stop_point:stopB",
+                    arrival=tstamp("20120615T000100"),
+                    departure=tstamp("20120615T180100"),
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival_skipped=True,
+                    departure_skipped=True,
+                    is_detour=True,
+                    message='deleted for detour',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopC",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    arrival=tstamp("20120616T003000"),
+                    departure=tstamp("20120616T003000"),
+                    is_added=True,
+                    is_detour=True,
+                    message='added for detour',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopA",
+                    arrival=tstamp("20120616T010102"),
+                    departure=tstamp("20120616T010102"),
+                    arrival_delay=7 * 60 * 60,
+                    message="No more pass midnight",
+                ),
+            ],
+            disruption_id='stop_time_with_detour',
+            effect='delayed',
+        )
+
+        # The disruption created above is modified so no disruption is added
+        disruptions_after = self.query_region(disruption_query)
+        assert initial_nb_disruptions + 1 == len(disruptions_after['disruptions'])
+
+        # Again we have 8 vehicle_journeys
```

Comment:
minor (if you stop checking the exact number of VJ at the beginning):
```suggestion
        # The disruption created above is modified so no vehicle_journey is added
```